In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression, LassoCV
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error
from sklearn.preprocessing import StandardScaler

import tensorflow as tf
from tensorflow.keras.models import Sequential
from tensorflow.keras.layers import Dense, Dropout
from tensorflow.keras.activations import relu, linear
from tensorflow.keras.losses import MeanAbsoluteError
from tensorflow.keras.optimizers.legacy import RMSprop

In [2]:
Gitters = pd.read_csv('../data/hitters.csv')
Gitters.head()

AtBat  Hits  HmRun  Runs  RBI  Walks  Years  CAtBat  CHits  CHmRun  CRuns  \
0    293    66      1    30   29     14      1     293     66       1     30   
1    315    81      7    24   38     39     14    3449    835      69    321   
2    479   130     18    66   72     76      3    1624    457      63    224   
3    496   141     20    65   78     37     11    5628   1575     225    828   
4    321    87     10    39   42     30      2     396    101      12     48   

   CRBI  CWalks League Division  PutOuts  Assists  Errors  Salary NewLeague  
0    29      14      A        E      446       33      20     NaN         A  
1   414     375      N        W      632       43      10   475.0         N  
2   266     263      A        W      880       82      14   480.0         A  
3   838     354      N        E      200       11       3   500.0         N  
4    46      33      N        E      805       40       4    91.5         N

In [3]:
Gitters = pd.get_dummies(Gitters)
Gitters.drop(['League_N', 'Division_E', 'NewLeague_N'], axis=1, inplace=True)
Gitters.dropna(inplace=True)

In [4]:
X = Gitters.drop(['Salary'], axis=1)
y = Gitters.Salary

n, m = X.shape

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=0)

In [5]:
# Fit a linear regression model to hitters data
lfit = LinearRegression(fit_intercept=True)
lfit.fit(X_train, y_train)

# Compute the mean absolute error
np.mean(np.absolute(y_test-lfit.predict(X_test)))

267.07095551089486

In [6]:
# Fit lasso regression with cross-validation
lassocv = LassoCV(cv=10, random_state=0, max_iter=10000).fit(X_train, y_train)

np.mean(np.absolute(y_test-lassocv.predict(X_test)))

263.5245245666904

In [7]:
# Fit a neural network 
model = Sequential(
    [
        Dense(units=50, activation='relu', input_shape=(m, )),
        Dropout(0.4),
        Dense(units=1)
    ], name='hitters_model'
)

model.summary()

Model: "hitters_model"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense (Dense)               (None, 50)                1000      
                                                                 
 dropout (Dropout)           (None, 50)                0         
                                                                 
 dense_1 (Dense)             (None, 1)                 51        
                                                                 
Total params: 1051 (4.11 KB)
Trainable params: 1051 (4.11 KB)
Non-trainable params: 0 (0.00 Byte)
_________________________________________________________________


In [8]:
# scaling input data

scaler = StandardScaler()
X = scaler.fit_transform(X)

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=0)

In [9]:
model.compile(
    loss=MeanAbsoluteError(),
    optimizer=RMSprop(),
    metrics=[MeanAbsoluteError()]
)

history = model.fit(X_train, y_train, epochs=1500, 
                    batch_size=32, validation_data=(X_test, y_test))

Epoch 1/1500
6/6 [==============================] - 0s 11ms/step - loss: 519.6552 - mean_absolute_error: 520.4284 - val_loss: 567.5153 - val_mean_absolute_error: 569.2108
Epoch 2/1500
6/6 [==============================] - 0s 2ms/step - loss: 519.2000 - mean_absolute_error: 522.6296 - val_loss: 567.1802 - val_mean_absolute_error: 568.8807
Epoch 3/1500
6/6 [==============================] - 0s 2ms/step - loss: 518.9643 - mean_absolute_error: 508.5414 - val_loss: 566.8793 - val_mean_absolute_error: 568.5844
Epoch 4/1500
6/6 [==============================] - 0s 2ms/step - loss: 518.6442 - mean_absolute_error: 526.5467 - val_loss: 566.6088 - val_mean_absolute_error: 568.3176
Epoch 5/1500
6/6 [==============================] - 0s 2ms/step - loss: 518.3080 - mean_absolute_error: 517.6979 - val_loss: 566.3425 - val_mean_absolute_error: 568.0544
Epoch 6/1500
6/6 [==============================] - 0s 2ms/step - loss: 518.1751 - mean_absolute_error: 521.3299 - val_loss: 566.0714 - val_mean_abso

Epoch 49/1500
6/6 [==============================] - 0s 2ms/step - loss: 501.4925 - mean_absolute_error: 514.1757 - val_loss: 548.2120 - val_mean_absolute_error: 550.1880
Epoch 50/1500
6/6 [==============================] - 0s 2ms/step - loss: 500.1652 - mean_absolute_error: 518.6221 - val_loss: 547.5806 - val_mean_absolute_error: 549.5673
Epoch 51/1500
6/6 [==============================] - 0s 2ms/step - loss: 499.7526 - mean_absolute_error: 501.1418 - val_loss: 546.9511 - val_mean_absolute_error: 548.9477
Epoch 52/1500
6/6 [==============================] - 0s 2ms/step - loss: 500.0051 - mean_absolute_error: 501.5081 - val_loss: 546.3305 - val_mean_absolute_error: 548.3376
Epoch 53/1500
6/6 [==============================] - 0s 2ms/step - loss: 498.9008 - mean_absolute_error: 494.6304 - val_loss: 545.6891 - val_mean_absolute_error: 547.7057
Epoch 54/1500
6/6 [==============================] - 0s 2ms/step - loss: 498.6578 - mean_absolute_error: 495.2751 - val_loss: 545.0254 - val_mean

Epoch 97/1500
6/6 [==============================] - 0s 2ms/step - loss: 468.5219 - mean_absolute_error: 487.7081 - val_loss: 513.5771 - val_mean_absolute_error: 515.6902
Epoch 98/1500
6/6 [==============================] - 0s 2ms/step - loss: 466.3405 - mean_absolute_error: 472.5594 - val_loss: 512.8799 - val_mean_absolute_error: 514.9772
Epoch 99/1500
6/6 [==============================] - 0s 2ms/step - loss: 466.3163 - mean_absolute_error: 465.0466 - val_loss: 512.1414 - val_mean_absolute_error: 514.2226
Epoch 100/1500
6/6 [==============================] - 0s 2ms/step - loss: 464.7522 - mean_absolute_error: 468.3051 - val_loss: 511.4741 - val_mean_absolute_error: 513.5447
Epoch 101/1500
6/6 [==============================] - 0s 2ms/step - loss: 464.2716 - mean_absolute_error: 466.8429 - val_loss: 510.8255 - val_mean_absolute_error: 512.8880
Epoch 102/1500
6/6 [==============================] - 0s 2ms/step - loss: 462.7392 - mean_absolute_error: 453.1034 - val_loss: 510.1053 - val_m

Epoch 145/1500
6/6 [==============================] - 0s 2ms/step - loss: 435.4398 - mean_absolute_error: 441.8123 - val_loss: 478.8557 - val_mean_absolute_error: 480.9382
Epoch 146/1500
6/6 [==============================] - 0s 2ms/step - loss: 431.8607 - mean_absolute_error: 421.6581 - val_loss: 478.0434 - val_mean_absolute_error: 480.1466
Epoch 147/1500
6/6 [==============================] - 0s 2ms/step - loss: 432.0673 - mean_absolute_error: 424.7602 - val_loss: 477.3190 - val_mean_absolute_error: 479.4344
Epoch 148/1500
6/6 [==============================] - 0s 2ms/step - loss: 429.6270 - mean_absolute_error: 439.2002 - val_loss: 476.5313 - val_mean_absolute_error: 478.6635
Epoch 149/1500
6/6 [==============================] - 0s 2ms/step - loss: 431.4869 - mean_absolute_error: 440.3650 - val_loss: 475.7479 - val_mean_absolute_error: 477.8936
Epoch 150/1500
6/6 [==============================] - 0s 2ms/step - loss: 430.8102 - mean_absolute_error: 432.7620 - val_loss: 475.0192 - va

Epoch 240/1500
6/6 [==============================] - 0s 2ms/step - loss: 366.6828 - mean_absolute_error: 367.0384 - val_loss: 396.8819 - val_mean_absolute_error: 400.9519
Epoch 241/1500
6/6 [==============================] - 0s 2ms/step - loss: 364.5699 - mean_absolute_error: 365.0761 - val_loss: 395.9036 - val_mean_absolute_error: 400.0006
Epoch 242/1500
6/6 [==============================] - 0s 2ms/step - loss: 365.6791 - mean_absolute_error: 363.9716 - val_loss: 394.9960 - val_mean_absolute_error: 399.1162
Epoch 243/1500
6/6 [==============================] - 0s 2ms/step - loss: 365.8928 - mean_absolute_error: 376.1041 - val_loss: 393.9740 - val_mean_absolute_error: 398.1260
Epoch 244/1500
6/6 [==============================] - 0s 2ms/step - loss: 364.0540 - mean_absolute_error: 357.5133 - val_loss: 393.0243 - val_mean_absolute_error: 397.1998
Epoch 245/1500
6/6 [==============================] - 0s 2ms/step - loss: 358.6131 - mean_absolute_error: 347.8510 - val_loss: 392.0674 - va

Epoch 335/1500
6/6 [==============================] - 0s 2ms/step - loss: 303.7365 - mean_absolute_error: 306.8179 - val_loss: 326.1219 - val_mean_absolute_error: 330.3183
Epoch 336/1500
6/6 [==============================] - 0s 2ms/step - loss: 302.9222 - mean_absolute_error: 308.1704 - val_loss: 325.4071 - val_mean_absolute_error: 329.5858
Epoch 337/1500
6/6 [==============================] - 0s 2ms/step - loss: 296.8399 - mean_absolute_error: 292.3363 - val_loss: 324.9986 - val_mean_absolute_error: 329.1639
Epoch 338/1500
6/6 [==============================] - 0s 2ms/step - loss: 295.9356 - mean_absolute_error: 296.7555 - val_loss: 324.5052 - val_mean_absolute_error: 328.6544
Epoch 339/1500
6/6 [==============================] - 0s 2ms/step - loss: 297.2713 - mean_absolute_error: 296.9916 - val_loss: 324.0252 - val_mean_absolute_error: 328.1631
Epoch 340/1500
6/6 [==============================] - 0s 2ms/step - loss: 298.9976 - mean_absolute_error: 299.5390 - val_loss: 323.4745 - va

Epoch 430/1500
6/6 [==============================] - 0s 2ms/step - loss: 262.6667 - mean_absolute_error: 258.8287 - val_loss: 292.4673 - val_mean_absolute_error: 294.8373
Epoch 431/1500
6/6 [==============================] - 0s 2ms/step - loss: 269.6497 - mean_absolute_error: 268.9521 - val_loss: 292.1626 - val_mean_absolute_error: 294.5129
Epoch 432/1500
6/6 [==============================] - 0s 2ms/step - loss: 257.7065 - mean_absolute_error: 256.7751 - val_loss: 291.8794 - val_mean_absolute_error: 294.2070
Epoch 433/1500
6/6 [==============================] - 0s 2ms/step - loss: 278.3466 - mean_absolute_error: 280.9005 - val_loss: 291.6462 - val_mean_absolute_error: 293.9586
Epoch 434/1500
6/6 [==============================] - 0s 2ms/step - loss: 259.9605 - mean_absolute_error: 262.0814 - val_loss: 291.3390 - val_mean_absolute_error: 293.6340
Epoch 435/1500
6/6 [==============================] - 0s 2ms/step - loss: 274.2843 - mean_absolute_error: 281.2118 - val_loss: 291.1022 - va

Epoch 525/1500
6/6 [==============================] - 0s 2ms/step - loss: 241.9023 - mean_absolute_error: 243.0399 - val_loss: 271.3891 - val_mean_absolute_error: 272.2738
Epoch 526/1500
6/6 [==============================] - 0s 2ms/step - loss: 247.6436 - mean_absolute_error: 250.3158 - val_loss: 271.1536 - val_mean_absolute_error: 272.0234
Epoch 527/1500
6/6 [==============================] - 0s 2ms/step - loss: 243.3619 - mean_absolute_error: 240.1287 - val_loss: 270.9733 - val_mean_absolute_error: 271.8302
Epoch 528/1500
6/6 [==============================] - 0s 2ms/step - loss: 228.6856 - mean_absolute_error: 234.5074 - val_loss: 270.8207 - val_mean_absolute_error: 271.6661
Epoch 529/1500
6/6 [==============================] - 0s 2ms/step - loss: 252.2910 - mean_absolute_error: 261.4142 - val_loss: 270.5468 - val_mean_absolute_error: 271.3760
Epoch 530/1500
6/6 [==============================] - 0s 2ms/step - loss: 248.6983 - mean_absolute_error: 246.8456 - val_loss: 270.3599 - va

Epoch 620/1500
6/6 [==============================] - 0s 2ms/step - loss: 245.5291 - mean_absolute_error: 235.3051 - val_loss: 261.3757 - val_mean_absolute_error: 261.1990
Epoch 621/1500
6/6 [==============================] - 0s 2ms/step - loss: 244.4218 - mean_absolute_error: 250.3617 - val_loss: 261.3103 - val_mean_absolute_error: 261.1272
Epoch 622/1500
6/6 [==============================] - 0s 2ms/step - loss: 249.8522 - mean_absolute_error: 247.6716 - val_loss: 261.1993 - val_mean_absolute_error: 260.9994
Epoch 623/1500
6/6 [==============================] - 0s 2ms/step - loss: 238.4989 - mean_absolute_error: 235.4460 - val_loss: 261.1518 - val_mean_absolute_error: 260.9474
Epoch 624/1500
6/6 [==============================] - 0s 2ms/step - loss: 246.2363 - mean_absolute_error: 243.0407 - val_loss: 261.0930 - val_mean_absolute_error: 260.8861
Epoch 625/1500
6/6 [==============================] - 0s 2ms/step - loss: 233.9236 - mean_absolute_error: 229.0440 - val_loss: 260.9575 - va

Epoch 715/1500
6/6 [==============================] - 0s 2ms/step - loss: 231.1934 - mean_absolute_error: 226.1936 - val_loss: 256.1830 - val_mean_absolute_error: 254.9998
Epoch 716/1500
6/6 [==============================] - 0s 2ms/step - loss: 227.5549 - mean_absolute_error: 223.3244 - val_loss: 256.0643 - val_mean_absolute_error: 254.8438
Epoch 717/1500
6/6 [==============================] - 0s 2ms/step - loss: 245.2505 - mean_absolute_error: 239.0654 - val_loss: 256.0776 - val_mean_absolute_error: 254.8623
Epoch 718/1500
6/6 [==============================] - 0s 2ms/step - loss: 226.6143 - mean_absolute_error: 225.5801 - val_loss: 256.0840 - val_mean_absolute_error: 254.8713
Epoch 719/1500
6/6 [==============================] - 0s 2ms/step - loss: 214.1975 - mean_absolute_error: 215.7085 - val_loss: 255.9849 - val_mean_absolute_error: 254.7440
Epoch 720/1500
6/6 [==============================] - 0s 2ms/step - loss: 227.4425 - mean_absolute_error: 226.1288 - val_loss: 255.9688 - va

Epoch 810/1500
6/6 [==============================] - 0s 2ms/step - loss: 222.4949 - mean_absolute_error: 219.2989 - val_loss: 253.7406 - val_mean_absolute_error: 251.7966
Epoch 811/1500
6/6 [==============================] - 0s 2ms/step - loss: 233.7009 - mean_absolute_error: 236.0842 - val_loss: 253.7316 - val_mean_absolute_error: 251.8045
Epoch 812/1500
6/6 [==============================] - 0s 2ms/step - loss: 222.5923 - mean_absolute_error: 219.5779 - val_loss: 253.7290 - val_mean_absolute_error: 251.8002
Epoch 813/1500
6/6 [==============================] - 0s 2ms/step - loss: 221.7649 - mean_absolute_error: 231.8246 - val_loss: 253.6864 - val_mean_absolute_error: 251.7377
Epoch 814/1500
6/6 [==============================] - 0s 2ms/step - loss: 222.2467 - mean_absolute_error: 220.8655 - val_loss: 253.6414 - val_mean_absolute_error: 251.6778
Epoch 815/1500
6/6 [==============================] - 0s 2ms/step - loss: 226.7995 - mean_absolute_error: 232.4861 - val_loss: 253.6882 - va

Epoch 905/1500
6/6 [==============================] - 0s 2ms/step - loss: 212.6326 - mean_absolute_error: 208.0825 - val_loss: 251.8980 - val_mean_absolute_error: 249.2043
Epoch 906/1500
6/6 [==============================] - 0s 2ms/step - loss: 222.0702 - mean_absolute_error: 214.5426 - val_loss: 251.9190 - val_mean_absolute_error: 249.2372
Epoch 907/1500
6/6 [==============================] - 0s 2ms/step - loss: 223.6476 - mean_absolute_error: 230.6513 - val_loss: 251.9413 - val_mean_absolute_error: 249.2629
Epoch 908/1500
6/6 [==============================] - 0s 2ms/step - loss: 202.7854 - mean_absolute_error: 198.6587 - val_loss: 251.8723 - val_mean_absolute_error: 249.1868
Epoch 909/1500
6/6 [==============================] - 0s 2ms/step - loss: 219.0906 - mean_absolute_error: 224.0384 - val_loss: 251.8835 - val_mean_absolute_error: 249.1951
Epoch 910/1500
6/6 [==============================] - 0s 2ms/step - loss: 213.6638 - mean_absolute_error: 207.0997 - val_loss: 251.9034 - va

Epoch 1000/1500
6/6 [==============================] - 0s 2ms/step - loss: 218.5225 - mean_absolute_error: 218.5865 - val_loss: 251.5300 - val_mean_absolute_error: 248.5401
Epoch 1001/1500
6/6 [==============================] - 0s 2ms/step - loss: 219.3012 - mean_absolute_error: 222.8352 - val_loss: 251.5101 - val_mean_absolute_error: 248.4818
Epoch 1002/1500
6/6 [==============================] - 0s 2ms/step - loss: 221.5704 - mean_absolute_error: 220.4452 - val_loss: 251.4628 - val_mean_absolute_error: 248.4389
Epoch 1003/1500
6/6 [==============================] - 0s 2ms/step - loss: 216.6969 - mean_absolute_error: 231.1440 - val_loss: 251.4744 - val_mean_absolute_error: 248.4384
Epoch 1004/1500
6/6 [==============================] - 0s 2ms/step - loss: 217.0602 - mean_absolute_error: 229.4634 - val_loss: 251.4885 - val_mean_absolute_error: 248.4121
Epoch 1005/1500
6/6 [==============================] - 0s 2ms/step - loss: 214.7092 - mean_absolute_error: 215.1490 - val_loss: 251.430

6/6 [==============================] - 0s 2ms/step - loss: 204.1197 - mean_absolute_error: 202.1180 - val_loss: 251.2498 - val_mean_absolute_error: 247.7367
Epoch 1095/1500
6/6 [==============================] - 0s 2ms/step - loss: 218.2322 - mean_absolute_error: 228.2884 - val_loss: 251.2497 - val_mean_absolute_error: 247.7332
Epoch 1096/1500
6/6 [==============================] - 0s 2ms/step - loss: 217.7109 - mean_absolute_error: 229.8920 - val_loss: 251.3188 - val_mean_absolute_error: 247.8259
Epoch 1097/1500
6/6 [==============================] - 0s 2ms/step - loss: 217.7205 - mean_absolute_error: 223.5997 - val_loss: 251.3144 - val_mean_absolute_error: 247.7988
Epoch 1098/1500
6/6 [==============================] - 0s 2ms/step - loss: 211.2231 - mean_absolute_error: 212.4057 - val_loss: 251.3009 - val_mean_absolute_error: 247.7764
Epoch 1099/1500
6/6 [==============================] - 0s 2ms/step - loss: 219.3751 - mean_absolute_error: 214.4037 - val_loss: 251.2945 - val_mean_abs

6/6 [==============================] - 0s 2ms/step - loss: 213.6584 - mean_absolute_error: 207.9017 - val_loss: 251.5488 - val_mean_absolute_error: 247.7612
Epoch 1189/1500
6/6 [==============================] - 0s 2ms/step - loss: 217.4194 - mean_absolute_error: 214.0912 - val_loss: 251.4587 - val_mean_absolute_error: 247.6693
Epoch 1190/1500
6/6 [==============================] - 0s 2ms/step - loss: 212.1108 - mean_absolute_error: 210.7501 - val_loss: 251.4741 - val_mean_absolute_error: 247.6931
Epoch 1191/1500
6/6 [==============================] - 0s 2ms/step - loss: 216.0729 - mean_absolute_error: 228.8247 - val_loss: 251.4385 - val_mean_absolute_error: 247.6786
Epoch 1192/1500
6/6 [==============================] - 0s 2ms/step - loss: 218.0476 - mean_absolute_error: 214.3678 - val_loss: 251.4559 - val_mean_absolute_error: 247.6929
Epoch 1193/1500
6/6 [==============================] - 0s 2ms/step - loss: 219.1625 - mean_absolute_error: 219.0466 - val_loss: 251.4371 - val_mean_abs

6/6 [==============================] - 0s 2ms/step - loss: 214.2982 - mean_absolute_error: 217.8355 - val_loss: 251.5704 - val_mean_absolute_error: 247.6413
Epoch 1283/1500
6/6 [==============================] - 0s 2ms/step - loss: 201.1940 - mean_absolute_error: 207.1277 - val_loss: 251.5408 - val_mean_absolute_error: 247.6001
Epoch 1284/1500
6/6 [==============================] - 0s 2ms/step - loss: 208.5747 - mean_absolute_error: 211.8113 - val_loss: 251.5757 - val_mean_absolute_error: 247.6440
Epoch 1285/1500
6/6 [==============================] - 0s 2ms/step - loss: 199.2219 - mean_absolute_error: 195.8340 - val_loss: 251.5883 - val_mean_absolute_error: 247.6485
Epoch 1286/1500
6/6 [==============================] - 0s 2ms/step - loss: 209.3075 - mean_absolute_error: 205.7875 - val_loss: 251.6297 - val_mean_absolute_error: 247.7121
Epoch 1287/1500
6/6 [==============================] - 0s 2ms/step - loss: 206.5244 - mean_absolute_error: 198.6595 - val_loss: 251.6024 - val_mean_abs

6/6 [==============================] - 0s 2ms/step - loss: 204.7713 - mean_absolute_error: 201.3224 - val_loss: 251.5305 - val_mean_absolute_error: 247.4683
Epoch 1377/1500
6/6 [==============================] - 0s 2ms/step - loss: 202.3441 - mean_absolute_error: 201.0358 - val_loss: 251.5498 - val_mean_absolute_error: 247.5004
Epoch 1378/1500
6/6 [==============================] - 0s 2ms/step - loss: 216.5992 - mean_absolute_error: 213.5078 - val_loss: 251.5715 - val_mean_absolute_error: 247.5557
Epoch 1379/1500
6/6 [==============================] - 0s 3ms/step - loss: 214.1845 - mean_absolute_error: 218.1068 - val_loss: 251.5591 - val_mean_absolute_error: 247.5394
Epoch 1380/1500
6/6 [==============================] - 0s 2ms/step - loss: 211.3265 - mean_absolute_error: 210.8395 - val_loss: 251.6732 - val_mean_absolute_error: 247.6510
Epoch 1381/1500
6/6 [==============================] - 0s 2ms/step - loss: 206.2346 - mean_absolute_error: 207.7589 - val_loss: 251.6596 - val_mean_abs

6/6 [==============================] - 0s 2ms/step - loss: 200.8181 - mean_absolute_error: 201.7040 - val_loss: 251.8481 - val_mean_absolute_error: 247.8326
Epoch 1471/1500
6/6 [==============================] - 0s 2ms/step - loss: 205.3658 - mean_absolute_error: 204.9124 - val_loss: 251.8238 - val_mean_absolute_error: 247.7962
Epoch 1472/1500
6/6 [==============================] - 0s 2ms/step - loss: 200.9456 - mean_absolute_error: 209.6336 - val_loss: 251.9003 - val_mean_absolute_error: 247.8964
Epoch 1473/1500
6/6 [==============================] - 0s 2ms/step - loss: 201.1636 - mean_absolute_error: 211.3310 - val_loss: 251.8835 - val_mean_absolute_error: 247.8676
Epoch 1474/1500
6/6 [==============================] - 0s 2ms/step - loss: 202.9696 - mean_absolute_error: 205.0457 - val_loss: 251.8422 - val_mean_absolute_error: 247.8154
Epoch 1475/1500
6/6 [==============================] - 0s 2ms/step - loss: 209.9963 - mean_absolute_error: 201.0281 - val_loss: 251.8490 - val_mean_abs

In [10]:
# Evaluate model on test data
model.evaluate(X_test, y_test)

3/3 [==============================] - 0s 761us/step - loss: 251.8200 - mean_absolute_error: 247.7599


[251.82003784179688, 247.7599334716797]

| Model                      | Mean Absolute Error |
|----------------------------|---------------------|
| Linear Regression          | 267.0710            |
| Lasso (with Cross-validation) | 263.52              |
| Neural Network (1 Hidden layer + Dropout Layer) | 247.7599            |